cpp/src/lists/drop_list_duplicates.cu

```diff
@@ -114,190 +118,160 @@ struct replace_negative_nans_fn {
 };
 
 /**
- * @brief A structure to be used along with type_dispatcher to replace -NaN by NaN for all rows
- * in a floating-point data column.
+ * @brief A structure to be used along with type_dispatcher to replace -NaN by NaN for a
+ * floating-point data column.
+ *
+ * Replacing -NaN by NaN is necessary before calling to segmented sort for lists because the sorting
+ * API may separate -NaN and NaN to the two ends of each result list while we want to group all NaN
+ * together.
```

Comment:
> Can all of this be replaced by normalize_nans_and_zeros?

No, it can only be partially replaced because `normalize_nans_and_zeros` expect a non-nested floating-point type. Maybe we can add extra work there. I'll file an issue.

Reply:
Done. Here it is: https://github.com/rapidsai/cudf/issues/9651